smsdb.c

```diff
@@ -241,8 +314,9 @@ EXPORT_DEF int smsdb_put(const char *id, const char *addr, int ref, int parts, i
 	char fullkey[MAX_DB_FIELD];
 	size_t fullkey_len;
 	int res = 0;
+	int ttl = CONF_GLOBAL(csms_ttl);
 
-	if (strlen(id) + strlen(addr) + 5 + 3 + 3 >= sizeof(fullkey)) {
+	if (strlen(id) + strlen(addr) + 3 + 3 + 3 >= sizeof(fullkey)) {
 		ast_log(LOG_WARNING, "Key length must be less than %zu bytes\n", sizeof(fullkey));
```

Comment:
That's not a WARNING, that's an ERROR, as the PUT is failing.

But.. why not do the if() _after_ the snprintf? (This manual addition is just bug-prone.)
Note that snprintf returns the number of characters _that would have been written if it was large enough_ (or <0 on other failure).